controller/src/main/java/io/pravega/controller/store/index/HostIndex.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.controller.store.index;
+
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Host Index provides a mechanism for tracking a bunch of entities managed by hosts in the cluster. Entities can
```

Comment:
Entity isn't a concept we use extensively, so I think we are trying to generalize here so that `HostIndex` can be used in multiple cases. Given the context of this PR, one example is txns. We need to clarify in any case what entities are here because it is not really clear as is.